src/wasm/literal.cpp

```diff
@@ -1070,6 +1070,14 @@ Literal Literal::subSatUI16(const Literal& other) const {
   return Literal(sub_sat_u<uint16_t>(geti32(), other.geti32()));
 }
 
+Literal Literal::q15MulrSatSI16(const Literal& other) const {
+  int64_t value =
+    (int64_t(geti32()) * int64_t(other.geti32()) + 0x4000LL) >> 15LL;
```

Comment:
Basically `int64_t` not necessary due to maximum value  of`0xFFFF * 0xFFFF + 0x4000` expression less than maximum uint32_t `4294967295`